client/my-sites/sharing/main.jsx

```diff
@@ -1,90 +1,114 @@
+	// verify that the required Jetpack module is active
+	if ( showConnections ) {
+		filters.push( {
+			id: 'sharing-connections',
+			route: '/sharing' + pathSuffix,
+			title: translate( 'Connections' ),
+		} );
+	}
 
-	getSelectedText: function() {
-		var selected = find( this.getFilters(), { path: this.props.path } );
-		if ( selected ) {
-			return selected.title;
-		}
+	// Include Sharing Buttons link if a site is selected and the
+	// required Jetpack module is active
+	if ( showButtons ) {
+		filters.push( {
+			id: 'sharing-buttons',
+			route: '/sharing/buttons' + pathSuffix,
+			title: translate( 'Sharing Buttons' ),
+		} );
+	}
 
-		return '';
-	},
+	const selected = find( filters, { route: path } );
 
-	getFilters: function() {
-		var site = sites.getSelectedSite(),
-			pathSuffix = sites.selected ? '/' + sites.selected : '',
-			filters = [];
+	return (
+		<Main className="sharing">
+			<DocumentHead title={ translate( 'Sharing', { textOnly: true } ) } />
+			{ siteId && <QueryJetpackModules siteId={ siteId } /> }
+			<SidebarNavigation />
+			{ filters.length > 0 &&
+				<SectionNav selectedText={ get( selected, 'title', '' ) }>
+					<NavTabs>
+						{ filters.map( ( { id, route, title } ) => (
+							<NavItem key={ id } path={ route } selected={ path === route }>
+								{ title }
+							</NavItem>
+						) ) }
+					</NavTabs>
+				</SectionNav>
+			}
+			<UpgradeNudge
+				event="sharing_no_ads"
+				feature="no-adverts"
+				message={ translate( 'Prevent ads from showing on your site.' ) }
+				title={ translate( 'No Ads with WordPress.com Premium' ) }
+			/>
+			{ contentComponent }
+		</Main>
+	);
+};
 
-		// Include Connections link if all sites are selected. Otherwise,
-		// verify that the required Jetpack module is active
-		if ( ! site || ! site.jetpack || site.isModuleActive( 'publicize' ) ) {
-			filters.push( { title: this.translate( 'Connections' ), path: '/sharing' + pathSuffix, id: 'sharing-connections' } );
-		}
+Sharing.propTypes = {
+	canManageOptions: PropTypes.bool,
+	contentComponent: PropTypes.node,
+	path: PropTypes.string,
+	showButtons: PropTypes.bool,
+	showConnections: PropTypes.bool,
+	siteId: PropTypes.number,
+	siteSlug: PropTypes.string,
+	translate: PropTypes.func,
+};
 
-		// Include Sharing Buttons link if a site is selected and the
-		// required Jetpack module is active
-		if ( site && utils.userCan( 'manage_options', site ) &&
-			( ! site.jetpack ||
-				( site.isModuleActive( 'sharedaddy' ) && site.versionCompare( '3.4-dev', '>=' ) )
-			)
-		) {
-			filters.push( { title: this.translate( 'Sharing Buttons' ), path: '/sharing/buttons' + pathSuffix, id: 'sharing-buttons' } );
-		}
+export default connect(
+	( state ) => {
+		const siteId = getSelectedSiteId( state );
+		const isJetpack = isJetpackSite( state, siteId );
 
-		return filters;
+		return {
+			showButtons: ( siteId && canCurrentUser( state, siteId, 'manage_options' ) && ( ! isJetpack || (
```

Comment:
This is quite the condition :D Not sure how else we might simplify it aside from maybe splitting off a few variables: `canManageOptions` ? `supportsSharingButtons` ?